lib/puppet/type/file/source.rb

```diff
@@ -322,7 +322,12 @@ def get_from_http_source(url, &block)
 
     def chunk_file_from_source(&block)
       if uri.scheme =~ /^https?/
-        get_from_http_source(uri, &block)
+        # Historically puppet has not encoded the http(s) source URL before parsing
+        # it, for example, if the path contains spaces, then it must be URL encoded
+        # as %20 in the manifest. Puppet behaves the same when retrieving file
+        # metadata via http(s), see Puppet::Indirector::FileMetadata::Http#find.
+        url = URI.parse(metadata.source)
```

Comment:
Should we use [URI.escape](https://apidock.com/ruby/URI/Escape) on `metadata.source` before parsing it, given that this is provided in the source parameter?

Reply:
Unfortunately, we can't/shouldn't as puppet 6.x assumes the caller has already URL escaped/encoded the source. If we were to escape/encode here, then it would end up being double-encoded.